### PR TITLE
message: add tests to increase code coverage

### PR DESCRIPTION
**Pull request by Danielius1922**

### CODE REVIEW

File: message/pool/message_test.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+func TestMessageSetPath(t *testing.T) {
+	msg := NewMessage()
+	size := 4
+	for i := 0; i < 8; i++ {
+		buffer := make([]byte, size)
+		buffer[0] = '/'
+		_, err := rand.Read(buffer[1:])
+		require.NoError(t, err)
+		inPath := string(buffer)
+		msg.SetPath(inPath)
```
Comment (Danielius1922) [inline message/pool/message_test.go:24]: SetPath crashes for paths longer than 255 chars,

[https://github.com/plgd-dev/go-coap/blob/c3e8d8de2f5c3a0cd69a2811200308613e6243bc/message/pool/message.go#L94-L99]

Comment (jkralik) [reply message/pool/message_test.go:24]: if `used < 0` then the error `message.ErrTooSmall` is incorrect. We need to introduce error `message.ErrTooLong` for this case.

File: message/pool/message.go [hunk 2]
```diff
@@ -90,19 +92,39 @@ func (r *Message) Options() message.Options {
...
+// (URI-Path segments must have maximal length of 255) the internal
+// buffer is expanded.
+// If the path is too long, but not valid then the function returns
+// ErrInvalidValueLength error.
+func (r *Message) SetPath(p string) error {
 	opts, used, err := r.msg.Options.SetPath(r.valueBuffer, p)
-
-	if err == message.ErrTooSmall {
-		r.valueBuffer = append(r.valueBuffer, make([]byte, used)...)
+	for err == message.ErrTooSmall {
```
Comment (Danielius1922) [inline message/pool/message.go:105]: NitPick: I could use something like

```go
expandBy := len(p) - len(r.valueBuffer)
```

and avoid the forcycle.

File: message/pool/message_test.go (hunk 1 above)
Comment (Danielius1922) [reply message/pool/message_test.go]: Should be fixed now. 

I've also found an edge case when the path had duplicated '/' characters. I've fixed that too and added tests.

**Review by jkralik — APPROVED** [review]
